Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -8169,21 +8169,18 @@ bool SpeculativeJIT::compileStrictEq(Node* node)
         return false;
     }
 
-    if (node->isBinaryUseKind(OtherUse)
+    if (node->isBinaryUseKind(BooleanUse, UntypedUse)
+        || node->isBinaryUseKind(UntypedUse, BooleanUse)
+        || node->isBinaryUseKind(OtherUse)
         || node->isBinaryUseKind(OtherUse, UntypedUse)
-        || node->isBinaryUseKind(UntypedUse, OtherUse)) {
-        compileOtherStrictEq(node);
-        return false;
-    }
-
-
-    if (node->isBinaryUseKind(MiscUse)
+        || node->isBinaryUseKind(UntypedUse, OtherUse)
+        || node->isBinaryUseKind(MiscUse)
         || node->isBinaryUseKind(MiscUse, UntypedUse)
         || node->isBinaryUseKind(UntypedUse, MiscUse)) {
-        compileMiscStrictEq(node);
+        compileBitwiseStrictEq(node);
```

Comment:
So, the rational here is that
```
MiscUse = null | undefined
UntypedUse = object | null | undefined
OtherUse = null | undefined
```
since JSNull and JSUndefined are unique in the implementation, we can just compare the JSValues.

Reply:
`MiscUse = Boolean | null | undefined` `UntypedUse` can be anything. But since `true` `false` `null` and `undefined` have exact representations we can do a bitwise comparison.